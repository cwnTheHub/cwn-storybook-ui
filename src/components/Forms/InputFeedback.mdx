import {
  Canvas,
  Meta,
  Title,
  Subtitle,
  Description,
  Primary,
  Controls,
  Stories,
  Story,
} from "@storybook/blocks";

import * as InputFeedbackStories from "./InputFeedback.stories";

<Meta of={InputFeedbackStories} />
<Title />

`import { InputFeedback } from '@nds_core/nds'`

### PROPS & METHODS

<Controls />

### Standalone usage
While its primary use is to facilitate feedback states for other form components such as Input, you may use it standalone.

<Canvas of={InputFeedbackStories.Standalone} />

### Advanced use with form components

Passing a function that returns an InputFeedback to the helper prop of form components gives you more control over the contents of the InputFeedback component by giving you access to the feedback state of the form field and the current value.

As an example, enter a value into the field below, then click away to lose focus. If you enter less than the 16 character minimum the helper will show as an error. Enter 16 or more characters to receive the success feedback.