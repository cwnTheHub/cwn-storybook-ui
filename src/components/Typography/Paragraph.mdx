import {
  Canvas,
  Meta,
  Title,
  Subtitle,
  Description,
  Primary,
  Controls,
  Stories,
  Story,
} from "@storybook/blocks";

import * as ParagraphStories from "./Paragraph.stories";

<Meta of={ParagraphStories} />
<Title />

`import {Paragraph} from '@nds_core/nds'`

Block text as an HTML `<p>` element.

### PROPS & METHODS

<Controls />

### Minimal usage

<Canvas of={ParagraphStories.MinimalUsage} />

### Usage criteria

- All body text should be contained in a `Paragraph`, regardless of length.
- If the `Paragraph` is on a dark background, the `invert` prop must be used to maintain sufficient colour contrast.
- If other modifications must be made to the text inside a `Paragraph` such as size or font weight, use the `Text`, `Small`, and `Strong` components.

### Placing text on dark colours

Use the invert prop.

<Story of={ParagraphStories.Inverted} />
