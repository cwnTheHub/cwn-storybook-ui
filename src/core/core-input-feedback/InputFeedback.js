import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { borders, messaging } from "../../shared-styles";
import Box from "../core-box/Box";
import { safeRest } from "../../util-helpers";

const StyledFeedback = styled(({ feedback, ...rest }) => <Box {...rest} />)(
  ({ feedback }) => ({
    ...borders.rounded,
    ...(feedback === "success" && messaging.success),
    ...(feedback === "error" && messaging.error),
    ...(feedback === undefined && messaging.standard),
  })
);

const InputFeedback = ({ feedback, children, ...rest }) => (
  <StyledFeedback
    {...safeRest(rest)}
    inset={3}
    role={feedback === "error" ? "alert" : null}
    feedback={feedback}
  >
    {children}
  </StyledFeedback>
);

InputFeedback.propTypes = {
  feedback: PropTypes.oneOf(["success", "error"]),
  children: PropTypes.node.isRequired,
};

InputFeedback.defaultProps = {
  feedback: undefined,
};

export default InputFeedback;
