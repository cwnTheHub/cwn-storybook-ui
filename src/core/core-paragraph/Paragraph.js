import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { DependentIconSizeContext, safeRest } from "../../util-helpers";
import * as typography from "../../shared-typography/typography";
import { spacing } from "../../shared-styles";

const paragraphColor = ({ invert }) =>
  invert ? typography.invertedColor : typography.color;
const paragraphInheritColor = ({ inheritColor }) =>
  inheritColor ? { color: "inherit" } : undefined;

const paragraphSize = ({ size }) => typography[size];
const paragraphBold = ({ bold, size }) =>
  bold ? typography.boldFont : typography[`${size}Font`];
const paragraphAlign = ({ align }) => ({
  textAlign: align,
});

export const StyledParagraph = styled.p(
  paragraphColor,
  typography.wordBreak,
  spacing.noSpacing,
  paragraphInheritColor,
  paragraphSize,
  paragraphBold,
  paragraphAlign,
  { sup: typography.sup }
);

const Paragraph = ({ size, invert, children, ...rest }, context) => {
  return (
    <DependentIconSizeContext.Provider value={{ paragraphSize: size, invert }}>
      <StyledParagraph
        {...safeRest(rest)}
        size={size}
        invert={invert}
        inheritColor={context.inheritColor}
      >
        {children}
      </StyledParagraph>
    </DependentIconSizeContext.Provider>
  );
};

Paragraph.propTypes = {
  bold: PropTypes.bool,
  size: PropTypes.oneOf(["small", "medium", "large"]),
  align: PropTypes.oneOf(["left", "center", "right"]),
  invert: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

Paragraph.defaultProps = {
  bold: false,
  size: "medium",
  align: "left",
  invert: false,
};

Paragraph.contextTypes = {
  inheritColor: PropTypes.bool,
};

export default Paragraph;
