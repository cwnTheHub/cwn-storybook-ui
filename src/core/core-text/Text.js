import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { DependentIconSizeContext, safeRest } from "../../util-helpers";
import * as typography from "../../shared-typography/typography";


const textColor = ({ invert }) =>
  invert ? typography.invertedColor : typography.color;
const textInheritColor = ({ inheritColor }) =>
  inheritColor ? { color: "inherit" } : undefined;
const textSize = ({ size }) => typography[size];
const textBold = ({ bold, size }) =>
  bold ? typography.boldFont : typography[`${size}Font`];
const textBlock = ({ block }) => (block ? typography.blockText : undefined);

// This named export is not guaranteed to be maintained and may be removed at any time.
export const StyledText = styled.span(
  textColor,
  textInheritColor,
  textSize,
  textBold,
  textBlock,
  {
    sup: typography.sup,
  }
);

const Text = ({ children, size, invert, ...rest }, context) => (
  <DependentIconSizeContext.Provider value={{ paragraphSize: size, invert }}>
    <StyledText
      {...safeRest(rest)}
      size={size}
      invert={invert}
      inheritColor={context.inheritColor}
    >
      {children}
    </StyledText>
  </DependentIconSizeContext.Provider>
);

Text.propTypes = {
  block: PropTypes.bool,
  bold: PropTypes.bool,
  size: PropTypes.oneOf(["base", "small", "medium", "large"]),
  invert: PropTypes.bool,
  children: PropTypes.node.isRequired,
};

Text.defaultProps = {
  block: false,
  bold: false,
  size: "base",
  invert: false,
};

Text.contextTypes = {
  inheritColor: PropTypes.bool,
};

export default Text;
