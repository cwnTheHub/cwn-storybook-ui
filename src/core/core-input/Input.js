import React, { useState } from "react";
import PropTypes from "prop-types";

import Box from "../core-box/Box";
import InputFeedback from "../core-input-feedback/InputFeedback";
import Text from "../core-text/Text";
import Paragraph from "../core-paragraph/Paragraph";
import { safeRest } from "../../util-helpers";
import FeedbackIcon from "./FeedbackIcon";
import {
  StyledFeedbackIcon,
  StyledInput,
  StyledLabelContainer,
} from "./styles";
import generateId from "../../utils/generateId";
import { componentWithName } from "../../util-prop-types";

const showFeedbackIcon = (feedback) =>
  feedback === "error" || feedback === "success";
const renderHint = (hint, Component, id) => (
  <Component id={id} size="small">
    {hint}
  </Component>
);
const renderError = (error, errorId) => (
  <InputFeedback id={errorId} feedback="error">
    <Paragraph size="small">{error}</Paragraph>
  </InputFeedback>
);
const renderLabel = (id, label, hint, hintPosition, hintId, tooltip) => (
  <div>
    <Box inline between={2}>
      <label htmlFor={id || generateId(label).identity()}>
        <StyledLabelContainer inline tag="span" between={2}>
          <Text size="medium" bold>
            {label}
          </Text>
          {hint && hintPosition === "inline" && renderHint(hint, Text, hintId)}
        </StyledLabelContainer>
      </label>
      {tooltip && React.cloneElement(tooltip, { connectedFieldLabel: label })}
    </Box>
    {hint && hintPosition === "below" && renderHint(hint, Paragraph, hintId)}
  </div>
);
const renderHelper = (helper, helperId, feedback, value) => {
  if (typeof helper === "function") {
    return (
      <div id={helperId}>
        <Text size="small">{helper(feedback, value)}</Text>
      </div>
    );
  }

  return (
    <InputFeedback id={helperId} feedback={feedback}>
      <Text size="small">{helper}</Text>
    </InputFeedback>
  );
};

const Input = React.forwardRef(
  (
    {
      id,
      value,
      type,
      label,
      hint,
      hintPosition,
      feedback,
      error,
      helper,
      tooltip,
      ...rest
    },
    ref
  ) => {
    const [isFocused, setIsFocused] = useState(false);
    const fieldId = generateId(id, rest.name, label);
    const errorId = error && fieldId.postfix("error-message");
    const helperId = helper && fieldId.postfix("helper");
    const hintId =
      (hint && hintPosition === "below" && fieldId.postfix("hint")) ||
      undefined;

    const handleFocus = (e) => {
      setIsFocused(true);
      if (rest.onFocus) {
        rest.onFocus(e);
      }
    };

    const handleBlur = (e) => {
      setIsFocused(false);
      if (rest.onBlur) {
        rest.onBlur(e);
      }
    };

    const handleKeyDown = (e) => {
      /**
       * this is a workaround for a bug in chrome that moves
       * the cursor into a wrong position if prepended with a space
       */
      if (type === "email" && e.key === " ") {
        e.preventDefault();
      }

      if (rest.onKeyDown) {
        rest.onKeyDown(e);
      }
    };

    return (
      <Box between={2}>
        {renderLabel(
          fieldId.identity(),
          label,
          hint,
          hintPosition,
          hintId,
          tooltip
        )}
        {helper && renderHelper(helper, helperId, feedback, value)}
        {error && renderError(error, errorId)}
        <div css={{ position: "relative" }}>
          <StyledInput
            {...safeRest(rest)}
            type={type}
            ref={ref}
            id={fieldId.identity()}
            value={value}
            feedback={feedback}
            aria-invalid={feedback === "error"}
            aria-describedby={errorId || hintId || helperId || undefined}
            onFocus={handleFocus}
            onBlur={handleBlur}
            onKeyDown={handleKeyDown}
          />
          {!rest.disabled && (
            <StyledFeedbackIcon>
              <FeedbackIcon
                showIcon={showFeedbackIcon(feedback) && !isFocused}
                feedback={feedback}
              />
            </StyledFeedbackIcon>
          )}
        </div>
      </Box>
    );
  }
);
Input.displayName = "Input";

Input.propTypes = {
  id: PropTypes.string,

  type: PropTypes.oneOf([
    "text",
    "number",
    "password",
    "email",
    "search",
    "tel",
    "url",
  ]),

  label: PropTypes.string.isRequired,

  hint: PropTypes.string,

  hintPosition: PropTypes.oneOf(["inline", "below"]),

  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),

  feedback: PropTypes.oneOf(["success", "error"]),

  error: PropTypes.node,

  helper: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),

  tooltip: componentWithName("Tooltip", true),
};

Input.defaultProps = {
  id: undefined,
  type: "text",
  hint: undefined,
  hintPosition: "inline",
  value: undefined,
  feedback: undefined,
  error: undefined,
  tooltip: undefined,
  helper: undefined,
};

export default Input;
