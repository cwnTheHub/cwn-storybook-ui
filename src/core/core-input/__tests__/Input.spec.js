import React from "react";
import { render, mount } from "enzyme";

import Input from "../Input";
import Tooltip from "../../core-tooltip/Tooltip";
import { Fade } from "../../../shared-animation";
import InputFeedback from "../../core-input-feedback/InputFeedback";
import Paragraph from "../../core-paragraph/Paragraph";
import Text from "../../core-text/Text";
import {
  NotificationSuccess,
  NotificationError,
} from "../../core-feedback-icon";

describe("Input", () => {
  const defaultProps = {
    label: "The input",
  };
  const doRender = (overrides = {}) =>
    render(<Input {...defaultProps} {...overrides} />);

  const doMount = (overrides = {}) => {
    const input = mount(<Input {...defaultProps} {...overrides} />);

    const findInputElement = () => input.find("input");

    return {
      input,
      label: input.find("label"),
      findFeedbackIconFade: () => input.find(Fade),
      findHelper: () => input.find(InputFeedback),
      findInputElement,
      changeValueTo: (value) =>
        findInputElement().simulate("change", { target: { value } }),
      focus: (focusEvent = {}) =>
        findInputElement().simulate("focus", focusEvent),
      blur: (blurEvent = {}) => findInputElement().simulate("blur", blurEvent),
    };
  };

  it("renders", () => {
    const input = doRender();

    expect(input).toMatchSnapshot();
  });

  it("renders with a feedback state and icon", () => {
    const input = doRender({ feedback: "error" });

    expect(input).toMatchSnapshot();
  });

  it("supports different input types", () => {
    let findInputElement = doMount().findInputElement;
    expect(findInputElement().props().type).toBe("text");

    findInputElement = doMount({ type: "number" }).findInputElement;
    expect(findInputElement().props().type).toBe("number");
  });

  describe("label", () => {
    it("must have a label", () => {
      const { label } = doMount({ label: "The label" });

      expect(
        label.contains(
          <Text size="medium" bold>
            The label
          </Text>
        )
      ).toBe(true);
    });

    it("can have a short hint", () => {
      const input = doMount({ id: "the-id", hint: "The short hint" });

      expect(input).toMatchSnapshot();
    });

    it("can have a long hint", () => {
      const { input, findInputElement } = doMount({
        id: "the-id",
        hint: "The long hint over here",
        hintPosition: "below",
      });

      expect(
        input.contains(
          <Paragraph id="the-id_hint" size="small">
            The long hint over here
          </Paragraph>
        )
      ).toBe(true);

      expect(findInputElement().props()["aria-describedby"]).toEqual(
        "the-id_hint"
      );
    });
  });

  describe("connecting the label to the input", () => {
    it("connects the label to the input", () => {
      const { label, findInputElement } = doMount();

      expect(label.prop("htmlFor")).toEqual(findInputElement().prop("id"));
    });

    it("uses the id when provided", () => {
      const { label, findInputElement } = doMount({
        id: "the-id",
        name: "the-name",
        label: "The label",
      });

      expect(label.props()["htmlFor"]).toBe("the-id");
      expect(findInputElement().props().id).toBe("the-id");
    });

    it("uses the name when no id is provided", () => {
      const { label, findInputElement } = doMount({
        name: "the-name",
        label: "The label",
      });

      expect(label.props()["htmlFor"]).toBe("the-name");
      expect(findInputElement().props().id).toBe("the-name");
    });

    it("generates an id from the label when no id or name is provided", () => {
      const { label, findInputElement } = doMount({ label: "The label" });

      expect(label.props()["htmlFor"]).toBe("the-label");
      expect(findInputElement().props().id).toBe("the-label");
    });
  });

  describe("editability", () => {
    it("supports string values or number values", () => {
      const onChangeMock = jest.fn();

      let findInputElement = doMount({
        value: "some value",
        readOnly: true,
      }).findInputElement;
      expect(findInputElement().getElement().props.value).toBe("some value");

      findInputElement = doMount({
        value: "another value",
        onChange: onChangeMock,
      }).findInputElement;
      expect(findInputElement().getElement().props.value).toBe("another value");

      findInputElement = doMount({
        value: 55,
        readOnly: true,
      }).findInputElement;
      expect(findInputElement().getElement().props.value).toBe(55);
    });

    it("will notify when its value changes", () => {
      const onChangeMock = jest.fn();

      const { changeValueTo } = doMount({ onChange: onChangeMock });
      changeValueTo("new value");

      expect(onChangeMock).toHaveBeenCalledWith(
        expect.objectContaining({ target: { value: "new value" } })
      );
    });

    it("can receive a new value from a parent component", () => {
      const { input, findInputElement } = doMount({
        value: "initial value",
        readOnly: true,
      });
      expect(findInputElement().getElement().props.value).toBe("initial value");

      input.setProps({ value: "new value" });

      expect(findInputElement().getElement().props.value).toBe("new value");
    });
  });

  describe("default values", () => {
    it("can be mounted with default value", () => {
      const { findInputElement } = doMount({ defaultValue: "initial value" });
      const input = findInputElement().getDOMNode();
      expect(input.value).toEqual("initial value");
    });
  });

  describe("feedback states", () => {
    it("renders a checkmark icon on success feedback state", () => {
      const { input } = doMount({ feedback: "success" });

      expect(input.find(NotificationSuccess)).toHaveLength(1);
    });

    it("renders an error icon on error feedback state", () => {
      const { input } = doMount({ feedback: "error" });

      expect(input.find(NotificationError)).toHaveLength(1);
    });

    it("fades the feedback icon in on focus lost and out on focus gained", () => {
      const { findFeedbackIconFade, focus } = doMount({ feedback: "success" });

      expect(findFeedbackIconFade().props().in).toBe(true);
      focus();
      expect(findFeedbackIconFade().props().in).toBe(false);
    });

    it("ensures that the contents do not overlap the icons", () => {
      const input = doRender({ feedback: "success" });

      expect(input).toMatchSnapshot();
    });
  });

  describe("disabling", () => {
    it("deactivates the input", () => {
      let findInputElement = doMount().findInputElement;
      expect(findInputElement().hasClass("disabled")).toBe(false);
      expect(findInputElement()).toBeDisabled;

      findInputElement = doMount({ disabled: true }).findInputElement;
      expect(findInputElement().hasClass("disabled")).toBe(false);
      expect(findInputElement()).toBeDisabled;
    });

    it("hides any icons", () => {
      const { input } = doMount({ disabled: true, feedback: "error" });

      expect(input.find(NotificationSuccess)).toHaveLength(0);
      expect(input.find(NotificationError)).toHaveLength(0);
    });
  });

  it("can have an error message", () => {
    const { input } = doMount({
      id: "some-id",
      error: "Oh no a terrible error!",
    });

    expect(
      input.contains(
        <InputFeedback id="some-id_error-message" feedback="error">
          <Paragraph size="small">Oh no a terrible error!</Paragraph>
        </InputFeedback>
      )
    ).toBe(true);
  });

  describe("helpers", () => {
    it("can have a simple helper of some components", () => {
      const helper = <div>Some helper text.</div>;
      const { input } = doMount({ id: "some-id", helper });

      expect(
        input.contains(
          <InputFeedback id="some-id_helper">
            <Text size="small">
              <div>Some helper text.</div>
            </Text>
          </InputFeedback>
        )
      ).toBe(true);
    });

    it("styles itself based on the feedback state", () => {
      const helper = <Paragraph>Some helper text.</Paragraph>;

      const { findHelper } = doMount({ feedback: "success", helper });

      expect(findHelper().props()["feedback"]).toEqual("success");
    });

    it("can have a complex helper function to give control to the consumer", () => {
      const helper = jest.fn();
      helper.mockReturnValue(<InputFeedback>Some helper text.</InputFeedback>);

      const { input } = doMount({
        id: "some-id",
        value: "current value",
        feedback: "error",
        readOnly: true,
        helper,
      });

      expect(helper).toHaveBeenCalledWith("error", "current value");
      expect(
        input.contains(
          <div id="some-id_helper">
            <Text size="small">
              <InputFeedback>Some helper text.</InputFeedback>
            </Text>
          </div>
        )
      ).toBe(true);
    });
  });

  describe("tooltip prop", () => {
    it("connects to Input", () => {
      const input = mount(
        <Input
          label="Some field"
          tooltip={<Tooltip copy="en">The tooltip</Tooltip>}
        />
      );

      expect(
        input.contains(
          <Tooltip copy="en" connectedFieldLabel="Some field">
            The tooltip
          </Tooltip>
        )
      ).toBe(true);
    });
  });

  describe("accessibility", () => {
    it("marks the input as invalid when in the error feedback state", () => {
      let findInputElement = doMount().findInputElement;
      expect(findInputElement().props()["aria-invalid"]).toEqual(false);

      findInputElement = doMount({ feedback: "error" }).findInputElement;
      expect(findInputElement().props()["aria-invalid"]).toEqual(true);
    });

    it("does not attach aria-describedby to the input field when no error or helper is present", () => {
      const { findInputElement } = doMount({
        error: undefined,
        helper: undefined,
      });

      expect(findInputElement().props()["aria-describedby"]).toEqual(undefined);
    });

    it("connects the error message to the input field for screen readers", () => {
      const { findInputElement, findHelper } = doMount({
        id: "some-field-id",
        error: "An error message",
      });

      expect(findInputElement().props()["aria-describedby"]).toEqual(
        "some-field-id_error-message"
      );
      expect(findHelper().props().id).toEqual("some-field-id_error-message");
    });

    it("connects a simple helper to the input field for screen readers", () => {
      const helper = <Paragraph>Some helper text.</Paragraph>;
      const { findInputElement, findHelper } = doMount({
        id: "some-field-id",
        helper,
      });

      expect(findInputElement().props()["aria-describedby"]).toEqual(
        "some-field-id_helper"
      );
      expect(findHelper().props().id).toEqual("some-field-id_helper");
    });

    it("connects a complex helper to the input field for screen readers", () => {
      const helper = () => <InputFeedback>Complex helper</InputFeedback>;
      const { input, findInputElement } = doMount({
        id: "some-field-id",
        helper,
      });

      expect(findInputElement().props()["aria-describedby"]).toEqual(
        "some-field-id_helper"
      );
      expect(
        input.contains(
          <div id="some-field-id_helper">
            <Text size="small">
              <InputFeedback>Complex helper</InputFeedback>
            </Text>
          </div>
        )
      ).toBe(true);
    });
  });

  describe("email type", () => {
    /**
     * This test is used to prevent regressions for a chrome bug that moves the cursor into a wrong
     * position if prepended with a space and the input type is email. It works by preventing any
     * keyDown events where the key is ' '.
     */
    it("prevents spaces", () => {
      const onKeyDownMock = jest.fn((e) => e.target.value);
      const preventDefaultMock = jest.fn();

      const { findInputElement } = doMount({
        label: "Email",
        type: "email",
        onKeyDown: onKeyDownMock,
      });
      const inputEl = findInputElement();
      expect(inputEl.props().type).toEqual("email");

      inputEl.simulate("keyDown", {
        key: " ",
        preventDefault: preventDefaultMock,
      });
      expect(preventDefaultMock).toHaveBeenCalled();
      expect(onKeyDownMock).toHaveReturned();
      preventDefaultMock.mockClear();
      onKeyDownMock.mockClear();

      inputEl.simulate("keyDown", {
        key: "j",
        preventDefault: preventDefaultMock,
      });
      expect(preventDefaultMock).not.toHaveBeenCalled();
      expect(onKeyDownMock).toHaveReturned();
    });
  });

  it("passes additional attributes to the input element", () => {
    const { findInputElement } = doMount({ name: "a name", id: "the-id" });

    expect(findInputElement().props().name).toEqual("a name");
    expect(findInputElement().props().id).toEqual("the-id");
  });

  it("does not allow custom CSS", () => {
    const { findInputElement } = doMount({
      className: "my-custom-class",
    });

    expect(findInputElement().hasClass("my-custom-class")).toBe(false);
  });
});
