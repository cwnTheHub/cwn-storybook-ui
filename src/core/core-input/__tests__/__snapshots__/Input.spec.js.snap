// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input feedback states ensures that the contents do not overlap the icons 1`] = `
<div
  class="sc-beqWaB dJZbbo"
>
  <div>
    <div
      class="sc-beqWaB vEPAP"
    >
      <label
        for="the-input"
      >
        <span
          class="sc-beqWaB vEPAP sc-iAEyYk feFWGD sc-iAEyYk feFWGD"
        >
          <span
            class="sc-dmqHEX hiTUaR"
          >
            The input
          </span>
        </span>
      </label>
    </div>
  </div>
  <div
    style="position:relative"
  >
    <input
      aria-invalid="false"
      class="sc-fsQiph hkYNsr"
      id="the-input"
      type="text"
    />
    <div
      class="sc-qRumB hXyFJJ"
    >
      <div
        style="transition:opacity 100ms ease-in-out;opacity:1"
      >
        <div
          class="sc-jsMahE kZMUa-d"
        >
          <svg
            class="sc-gLDzan eVwPjW"
            height="1.25rem"
            role="img"
            viewBox="0 0 20 20"
            width="1.25rem"
          >
            <title>
              The value of this input field is valid.
            </title>
            <path
              d="M0 10C0 4.48 4.48 0 10 0s10 4.48 10 10-4.48 10-10 10S0 15.52 0 10zm8.127 4.673a.633.633 0 0 0 .092-.105l7.734-8.572a.693.693 0 0 0-.009-.925.595.595 0 0 0-.882-.001l-7.514 8.055-2.612-2.8a.596.596 0 0 0-.88 0 .706.706 0 0 0-.014.928l3.038 3.51a.623.623 0 0 0 .486.219.587.587 0 0 0 .46-.2l.087-.097.014-.012z"
              fill="#2B8000"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Input label can have a short hint 1`] = `
Object {
  "blur": [Function],
  "changeValueTo": [Function],
  "findFeedbackIconFade": [Function],
  "findHelper": [Function],
  "findInputElement": [Function],
  "focus": [Function],
  "input": <Input
    hint="The short hint"
    hintPosition="inline"
    id="the-id"
    label="The input"
    type="text"
  >
    <Box
      between={2}
      inline={false}
      tag="div"
    >
      <styled.div
        between={2}
        inline={false}
        tag="div"
      >
        <div
          className="sc-beqWaB dJZbbo"
        >
          <div>
            <Box
              between={2}
              inline={true}
              tag="div"
            >
              <styled.div
                between={2}
                inline={true}
                tag="div"
              >
                <div
                  className="sc-beqWaB vEPAP"
                >
                  <label
                    htmlFor="the-id"
                  >
                    <Styled(Box)
                      between={2}
                      inline={true}
                      tag="span"
                    >
                      <Box
                        between={2}
                        className="sc-iAEyYk feFWGD"
                        inline={true}
                        tag="span"
                      >
                        <styled.div
                          between={2}
                          className="sc-iAEyYk feFWGD"
                          inline={true}
                          tag="span"
                        >
                          <span
                            className="sc-beqWaB vEPAP sc-iAEyYk feFWGD sc-iAEyYk feFWGD"
                          >
                            <Text
                              block={false}
                              bold={true}
                              invert={false}
                              size="medium"
                            >
                              <styled.span
                                block={false}
                                bold={true}
                                invert={false}
                                size="medium"
                              >
                                <span
                                  className="sc-dmqHEX hiTUaR"
                                  size="medium"
                                >
                                  The input
                                </span>
                              </styled.span>
                            </Text>
                            <Text
                              block={false}
                              bold={false}
                              invert={false}
                              size="small"
                            >
                              <styled.span
                                block={false}
                                bold={false}
                                invert={false}
                                size="small"
                              >
                                <span
                                  className="sc-dmqHEX dbentL"
                                  size="small"
                                >
                                  The short hint
                                </span>
                              </styled.span>
                            </Text>
                          </span>
                        </styled.div>
                      </Box>
                    </Styled(Box)>
                  </label>
                </div>
              </styled.div>
            </Box>
          </div>
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <styled.input
              aria-invalid={false}
              id="the-id"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              type="text"
            >
              <input
                aria-invalid={false}
                className="sc-fsQiph cdrdnB"
                id="the-id"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
              />
            </styled.input>
            <styled.div>
              <div
                className="sc-qRumB hXyFJJ"
              >
                <FeedbackIcon
                  showIcon={false}
                >
                  <Fade
                    in={false}
                    mountOnEnter={true}
                    timeout={100}
                    unmountOnExit={true}
                  >
                    <Transition
                      appear={false}
                      enter={true}
                      exit={true}
                      in={false}
                      mountOnEnter={true}
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={100}
                      unmountOnExit={true}
                    />
                  </Fade>
                </FeedbackIcon>
              </div>
            </styled.div>
          </div>
        </div>
      </styled.div>
    </Box>
  </Input>,
  "label": <label
    htmlFor="the-id"
  >
    <Styled(Box)
      between={2}
      inline={true}
      tag="span"
    >
      <Box
        between={2}
        className="sc-iAEyYk feFWGD"
        inline={true}
        tag="span"
      >
        <styled.div
          between={2}
          className="sc-iAEyYk feFWGD"
          inline={true}
          tag="span"
        >
          <span
            className="sc-beqWaB vEPAP sc-iAEyYk feFWGD sc-iAEyYk feFWGD"
          >
            <Text
              block={false}
              bold={true}
              invert={false}
              size="medium"
            >
              <styled.span
                block={false}
                bold={true}
                invert={false}
                size="medium"
              >
                <span
                  className="sc-dmqHEX hiTUaR"
                  size="medium"
                >
                  The input
                </span>
              </styled.span>
            </Text>
            <Text
              block={false}
              bold={false}
              invert={false}
              size="small"
            >
              <styled.span
                block={false}
                bold={false}
                invert={false}
                size="small"
              >
                <span
                  className="sc-dmqHEX dbentL"
                  size="small"
                >
                  The short hint
                </span>
              </styled.span>
            </Text>
          </span>
        </styled.div>
      </Box>
    </Styled(Box)>
  </label>,
}
`;

exports[`Input renders 1`] = `
<div
  class="sc-beqWaB dJZbbo"
>
  <div>
    <div
      class="sc-beqWaB vEPAP"
    >
      <label
        for="the-input"
      >
        <span
          class="sc-beqWaB vEPAP sc-iAEyYk feFWGD sc-iAEyYk feFWGD"
        >
          <span
            class="sc-dmqHEX hiTUaR"
          >
            The input
          </span>
        </span>
      </label>
    </div>
  </div>
  <div
    style="position:relative"
  >
    <input
      aria-invalid="false"
      class="sc-fsQiph cdrdnB"
      id="the-input"
      type="text"
    />
    <div
      class="sc-qRumB hXyFJJ"
    />
  </div>
</div>
`;

exports[`Input renders with a feedback state and icon 1`] = `
<div
  class="sc-beqWaB dJZbbo"
>
  <div>
    <div
      class="sc-beqWaB vEPAP"
    >
      <label
        for="the-input"
      >
        <span
          class="sc-beqWaB vEPAP sc-iAEyYk feFWGD sc-iAEyYk feFWGD"
        >
          <span
            class="sc-dmqHEX hiTUaR"
          >
            The input
          </span>
        </span>
      </label>
    </div>
  </div>
  <div
    style="position:relative"
  >
    <input
      aria-invalid="true"
      class="sc-fsQiph gHFngC"
      id="the-input"
      type="text"
    />
    <div
      class="sc-qRumB hXyFJJ"
    >
      <div
        style="transition:opacity 100ms ease-in-out;opacity:1"
      >
        <div
          class="sc-jsMahE kZMUa-d"
        >
          <svg
            class="sc-gLDzan eVwPjW"
            height="1.25rem"
            role="img"
            viewBox="0 0 20 20"
            width="1.25rem"
          >
            <title>
              The value of this input field is invalid.
            </title>
            <path
              d="M0 10C0 4.48 4.48 0 10 0s10 4.48 10 10-4.48 10-10 10S0 15.52 0 10zm10 5.833a.833.833 0 1 0 0-1.666.833.833 0 0 0 0 1.666zm-.003-2.5c.354 0 .644-.306.654-.69l.182-6.912c.01-.4-.285-.731-.654-.731H9.82c-.369 0-.664.33-.654.73l.175 6.912c.01.385.3.691.655.691z"
              fill="#c12335"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
